packages/currency/src/currencyManager.ts

```diff
@@ -71,17 +71,21 @@ export class CurrencyManager<TMeta = unknown> implements ICurrencyManager<TMeta>
       return this.fromAddress(currencyIdentifier, network);
     }
 
+    if (network && currencyIdentifier.indexOf(network) === -1) {
+      currencyIdentifier = `${currencyIdentifier}-${network}`;
```

Comment:
use the existing (static) method `currencyId`. You can change the signature to this: 

```ts
 static currencyId(currency: { symbol: string; network?: string }): string {
```